fix: XSS issues in page descriptions

Brookke opened the pull request:
uses bluemonday to sanitize html, since we are loading in that library I have also applied the same to the strip html. 

This did break a test turning `<p>1</p><p>2</p>` into `12` rather than `1\n\n2` I think this is an acceptable trade off since paragraphs should probably be on a new line anyway.

Brookke committed:
use bluemonday to sanitize html

markspolakovs commented on utils/template.go:
```diff
@@ -110,14 +111,6 @@ func RenderTemplate(w http.ResponseWriter, context structs.PageContext, data int
 			return t.Format(fmt)
 		},
-		// TODO(CaptainHayashi): this is temporary
-		"stripHTML": func(s string) string {
```
I assume you’ve checked this isn’t used anywhere?

Brookke replied:
Ah yes I have broken something, will fix that now 🤦🏼‍♂️, I thought it was re-declaration

Brookke replied:
oh bugger this new one doesnt handle `'` properly
![Screenshot 2021-09-19 at 19 35 41](https://user-images.githubusercontent.com/10200358/133938951-66146c26-5ae8-4560-ac5d-5f62eb453469.png)

Brookke replied:
Will take another look at this tomorrow, unless someone beats me to it :)

MattWindsor91 replied:
I look forward to any and all instances of hacky hayashicode being removed :o)

markspolakovs reviewed (commented):
Looks alright to me once you fix the build failure 

> This did break a test turning <p>1</p><p>2</p> into 12 rather than 1\n\n2 I think this is an acceptable trade off since paragraphs should probably be on a new line anyway.

I’d agree, though I’d be interested to see how prevalent this is in the wild (e.g. does MyRadio’s TinyMCE put in newlines or not)

Brookke committed:
update test

Brookke committed:
update schedule template

Brookke commented:
Just remembered this never got merged, I will circle back on this soon and lets get it in :)

Brookke commented:
> Looks alright to me once you fix the build failure
> 
> > This did break a test turning 12 into 12 rather than 1\n\n2 I think this is an acceptable trade off since paragraphs should probably be on a new line anyway.
> 
> I’d agree, though I’d be interested to see how prevalent this is in the wild (e.g. does MyRadio’s TinyMCE put in newlines or not)

One option here is to run the html through a formatter first

Brookke committed:
remove title

Brookke commented:
No idea who is best to review this these days, but I have rebased and made a few changes to get this into a state where it can be shipped.

_RE https://github.com/UniversityRadioYork/2016-site/pull/307#discussion_r711785620_

I have just removed the title tag, not sure its helpful to have the full show description in a little tooltip anyway

> This did break a test turning <p>1</p><p>2</p> into 12 rather than 1\n\n2 I think this is an acceptable trade off since paragraphs should probably be on a new line anyway.

RE the above - I have had a cursory glance at the current shows on the schedule and they all look like they are formatted correctly.

The value of this seems higher than the cost imo, finally closing out the XSS issue.
